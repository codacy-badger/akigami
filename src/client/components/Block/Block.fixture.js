import Block from './Block';

export default [
  {
    component: Block,
    name: 'no title',
    props: {
      children: 'lorem ipsum',
    },
  },
  {
    component: Block,
    name: 'with title',
    props: {
      title: 'Заголовок',
      children: 'lorem ipsum',
    },
  },
  {
    component: Block,
    name: 'bordered',
    props: {
      bordered: true,
      title: 'Заголовок',
      children: 'lorem ipsum',
    },
  },
  {
    component: Block,
    name: 'bordered & padded',
    props: {
      padded: true,
      bordered: true,
      title: 'Заголовок',
      children: 'lorem ipsum',
    },
  },
  {
    component: Block,
    name: 'colored & padded',
    props: {
      padded: true,
      colored: true,
      title: 'Заголовок',
      children: 'lorem ipsum',
    },
  },
  {
    component: Block,
    name: 'colored & padded & shadow',
    props: {
      padded: true,
      colored: true,
      shadow: true,
      title: 'Заголовок',
      children: 'lorem ipsum',
    },
  },
];
