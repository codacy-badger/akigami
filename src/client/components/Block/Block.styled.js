import { css } from 'emotion';
import styled from '@emotion/styled'

const bordered = css`
  border-radius: 2px;
  box-shadow: 0 0 0 1px rgba(0, 0, 0, 0.06);
`;

const colored = css`
  background-color: white;
`;

const padded = css`
  padding: 16px;
`;

const shadow = css`
  box-shadow: 0 0 0 1px rgba(0, 0, 0, 0.06), 0 7px 5px -5px rgba(0, 0, 0, 0.1);
`;

export const Wrapper = styled('div')`
  ${p => p.bordered && bordered} ${p => p.shadow && shadow};
  ${p => p.colored && colored};
  ${p => p.padded && padded};
  margin-bottom: 12px;
`;

export const Header = styled('div')`
  display: flex;
  align-items: center;
  margin-bottom: 16px;
`;

export const Title = styled('h3')`
  margin: 0;
  font-weight: bold;
  font-family: 'Proxima Nova', sans-serif;

  &:after {
    content: '';
    width: 40px;
    margin-top: 4px;
    height: 4px;
    background-color: #9b9b9b;
    display: block;
    border-radius: 4px;
  }

  @media screen and (max-width: 650px) {
    font-size: 20px;
  }
`;

export const Buttons = styled('div')`
  margin-left: auto;
`;

export const Content = styled('div')`
  display: flex;
  flex-wrap: wrap;
  flex-direction: ${p => (p.flex ? 'row' : 'column')};
  font-family: 'Proxima Nova', sans-serif;
`;
