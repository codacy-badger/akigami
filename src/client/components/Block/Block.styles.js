import styled from '@emotion/styled';

export default styled('div')`
  background: ${p => p.theme.colors.white};
  border-radius: ${p => p.theme.borderRadius};
  position: relative;
  margin-bottom: ${p => (p.mb ? '28px' : '0')};
  padding: ${p => (p.padded ? '16px 12px' : '0')};
  box-shadow: 0 2px 8px -4px ${p => p.theme.mixins.withOpacity(p.theme.colors.default, 0.4)};
  overflow: ${p => p.overflow};
`;
