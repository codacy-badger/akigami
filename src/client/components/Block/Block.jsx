import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Wrapper, Header, Title, Buttons, Content } from './Block.styled';

class Block extends PureComponent {
  static propTypes = {
    title: PropTypes.string,
    buttons: PropTypes.any,
    flex: PropTypes.bool,
    children: PropTypes.any,
    className: PropTypes.string,
    bordered: PropTypes.bool,
    shadow: PropTypes.bool,
    padded: PropTypes.bool,
    colored: PropTypes.bool,
  };
  static defaultProps = {
    title: null,
    children: null,
    buttons: null,
    className: null,
    flex: false,
    bordered: false,
    shadow: false,
    padded: false,
    colored: false,
  };
  render() {
    const {
      buttons,
      title,
      children,
      flex,
      className,
      bordered,
      padded,
      shadow,
      colored,
    } = this.props;
    return (
      <Wrapper
        bordered={bordered}
        colored={colored}
        padded={padded}
        shadow={shadow}
        className={className}
      >
        {title && (
          <Header>
            <Title>{title}</Title>
            {buttons && <Buttons>{buttons}</Buttons>}
          </Header>
        )}
        <Content flex={flex}>{children}</Content>
      </Wrapper>
    );
  }
}

export default Block;
