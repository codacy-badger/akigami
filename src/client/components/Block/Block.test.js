import { create as renderer } from 'react-test-renderer';
import createTestContext from 'react-cosmos-test/generic';
import Block from './Block.fixture';

Block.forEach((fixture) => {
  const res = createTestContext({ renderer, fixture });
  beforeEach(res.mount);
  test(`Block ${fixture.name} snapshot`, () => {
    expect(res.getWrapper().toJSON()).toMatchSnapshot();
  });
});

