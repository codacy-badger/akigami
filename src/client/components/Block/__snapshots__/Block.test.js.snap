// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block bordered & padded snapshot 1`] = `
<div
  className="css-nczllq"
>
  <div
    className="css-czr64w"
  >
    <h3
      className="css-1r0b13k"
    >
      Заголовок
    </h3>
  </div>
  <div
    className="css-bqvltc"
  >
    lorem ipsum
  </div>
</div>
`;

exports[`Block bordered snapshot 1`] = `
<div
  className="css-nczllq"
>
  <div
    className="css-czr64w"
  >
    <h3
      className="css-1r0b13k"
    >
      Заголовок
    </h3>
  </div>
  <div
    className="css-bqvltc"
  >
    lorem ipsum
  </div>
</div>
`;

exports[`Block colored & padded & shadow snapshot 1`] = `
<div
  className="css-nczllq"
>
  <div
    className="css-czr64w"
  >
    <h3
      className="css-1r0b13k"
    >
      Заголовок
    </h3>
  </div>
  <div
    className="css-bqvltc"
  >
    lorem ipsum
  </div>
</div>
`;

exports[`Block colored & padded snapshot 1`] = `
<div
  className="css-nczllq"
>
  <div
    className="css-czr64w"
  >
    <h3
      className="css-1r0b13k"
    >
      Заголовок
    </h3>
  </div>
  <div
    className="css-bqvltc"
  >
    lorem ipsum
  </div>
</div>
`;

exports[`Block no title snapshot 1`] = `
<div
  className="css-nczllq"
>
  <div
    className="css-czr64w"
  >
    <h3
      className="css-1r0b13k"
    >
      Заголовок
    </h3>
  </div>
  <div
    className="css-bqvltc"
  >
    lorem ipsum
  </div>
</div>
`;

exports[`Block with title snapshot 1`] = `
<div
  className="css-nczllq"
>
  <div
    className="css-czr64w"
  >
    <h3
      className="css-1r0b13k"
    >
      Заголовок
    </h3>
  </div>
  <div
    className="css-bqvltc"
  >
    lorem ipsum
  </div>
</div>
`;
